src/CSnakes.Runtime/Python/GIL.cs

```diff
@@ -26,25 +26,33 @@ namespace CSnakes.Runtime.Python;
 public static class GIL
 {
     [ThreadStatic] private static PyGilState? currentState;
+    [ThreadStatic] private static nint pythonThreadState;
     private static ConcurrentQueue<nint> handlesToDispose = new();
 
+    static GIL() {
+        currentState = null;
+        pythonThreadState = 0;
+    }
+
     internal class PyGilState : IDisposable
     {
-        private nint gilState;
         private int recursionCount;
 
         public PyGilState()
         {
+            if (pythonThreadState == 0){
+                pythonThreadState = CPythonAPI.PyThreadState_New();
+            }
             Debug.Assert(CPythonAPI.IsInitialized);
-            gilState = CPythonAPI.PyGILState_Ensure();
+            CPythonAPI.PyEval_RestoreThread(pythonThreadState);
             recursionCount = 1;
         }
 
         public void Ensure()
         {
             if (recursionCount == 0)

```

Comment:
In the alternate branch, we should expect there is a current thread state, right?  Would it be worth adding an assert here to that effect?

Would we also expect that it would always be `pythonThreadState`?